From de4c321bbc2f019775b55d95f575124d98026c84 Mon Sep 17 00:00:00 2001
From: Arockiasathish S <arocs@qti.qualcomm.com>
Date: Tue, 1 Sep 2026 18:50:20 +0530
Subject: [PATCH] [RDK][QMESH] replace arping gateway confirmation with a 
 bridge FDB + kernel neighbour table lookup; arping fails on bridge slave 
 ports because the ARP reply is intercepted by the bridge before the raw 
 socket sees it

Signed-off-by: Arockiasathish S <arocs@qti.qualcomm.com>
---
 .../src/beerocks/slave/backhaul_manager/backhaul_manager.cpp  | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
index 993985042..ea0105388 100644
--- a/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
+++ b/agent/src/beerocks/slave/backhaul_manager/backhaul_manager.cpp
@@ -2109,6 +2109,10 @@ bool BackhaulManager::arping_confirm_gateway(const std::string &iface, const std
 {
     LOG(INFO) << "Confirming GW IP (" << gw_ip << ") is reachable on " << iface;
 
+#ifdef BEEROCKS_RDKB
+    return beerocks::bpl::cfg_confirm_gateway_reachable(iface, gw_ip);
+#endif
+
     // Build arping command
     std::string arping_cmd = "arping -c 1 -I " + iface;
 
-- 
2.34.1

